scripts/analyze_data.py

```diff
@@ -0,0 +1,179 @@
+import os
+import argparse
+import json
+import glob
+from progress.bar import Bar
+import csv
+
+from utils import get_img_path_from_mask_path, get_mask_path_from_img_path, edit_metric_dict, save_graphs, change_mask_suffix, get_deriv_sub_from_img_path, str_to_float_list, str_to_str_list, mergedict
+
+def run_analysis(args):
```

Comment:
This function is slightly "heavy".
Would it be possible to split it into several smaller functions (10-20 lines per function)? Each one with a self-explaning docstring.
Also, some nested `for` loops and `if-else` conditions are hard to follow. Adding comments would make them easier to follow.